src/Classes/Entities/EventEntity.php

```diff
@@ -28,5 +28,179 @@ public function __construct(
         $this->startTime = ($this->startTime ?? $startTime);
         $this->endTime = ($this->endTime ?? $endTime);
         $this->notes = ($this->notes ?? $notes);
+
+        $this->sanitiseData();
+    }
+
+    /**
+     * Will sanitise all the fields for an event.
+     */
+    private function sanitiseData()
+    {
+        $this->name = $this->sanitiseString($this->name);
+        $this->name = self::validateExistsAndLength($this->name, 255);
+        $this->category = (int)$this->category;
+        $this->location = $this->sanitiseString($this->location);
+        $this->location = self::validateExistsAndLength($this->location, 255);
+        $this->date = $this->validateDate($this->date);
+        $this->startTime = $this->validateTime($this->startTime);
+        $this->endTime = $this->validateTime($this->endTime);
+        $this->notes = $this->sanitiseString($this->notes);
+        $this->notes = self::validateLength($this->notes, 255);
+    }
+
+    /**
+     * Sanitise as a date in the event table as data.
+     *
+     * @param $date
+     *
+     * @return bool|string
+     */
+    public function validateDate($date)
+    {
+        if (!preg_match('/([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/', $date)) {
+            return false;
```

Comment:
This will assign a value of false to the date property of the entity - what then happens to that value of false on the way to the database?

Reply:
changed to throw Exception